src/main/frontend/views/rebalance/ReBalanceView.tsx

```
@@ -0,0 +1,113 @@
+                value: fund.value,
+                ratio: fund.ratio / 100,  // Convert ratio to fraction
+            }));
+
+            // Call server-side method and get the InvestmentResponse
+            const response: InvestmentResponse | undefined = await reBalance({
+                funds: fundsData,
+                amountToInvest: Number(amountToInvest),
+            });
+
+            // Check if response and investments exist before processing
+            if (response && response.investments) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)